abi/dynamic/reflect_marshal.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package dynamic
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	"github.com/ava-labs/hypersdk/abi"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/consts"
+)
+
+func DynamicMarshal(inputAbi abi.ABI, typeName string, jsonData string) ([]byte, error) {
+	// Find the type in the ABI
+	abiType := findABIType(inputAbi, typeName)
+	if abiType == nil {
+		return nil, fmt.Errorf("type %s not found in ABI", typeName)
+	}
+
+	// Create a cache to avoid rebuilding types
+	typeCache := make(map[string]reflect.Type)
+
+	// Create a dynamic struct type
+	dynamicType := getReflectType(typeName, inputAbi, typeCache)
```

Comment:
I prefer using full English words for variable names and want to avoid abbreviations like `typ` instead of `type` just for aesthetics. However, since `type` is a reserved keyword, I came up with `dynamicType` as the best alternative.